aeneas/src/ir/SsaNormalizer.v3

```diff
@@ -1352,9 +1339,52 @@ class SsaRaNormalizer extends SsaRebuilder {
 			curBlock.opClassSetField(nf[i], receiver, ai_new[i + 1], init);
 		}
 	}
-	def normNullCheck(i_old: SsaApplyOp, op: Operator) {
-		var ai_new = genRefs(i_old.inputs);
-		if (ai_new.length >= 1) addNullCheck(i_old, ai_new[0]);
+	def genVariantScalarView(oldType: Type, newType: Type, i_scalar: SsaInstr) -> SsaInstr {
+		if (oldType == newType) return i_scalar;
+		if (IntType.?(oldType) && IntType.?(newType)) return curBlock.opIntView(oldType, IntType.!(newType), i_scalar);
+		if (IntType.?(oldType)) {
+			match (newType) {
+				x: BoolType => return curBlock.opNotEqual(oldType, i_scalar, newGraph.nullConst(oldType));
+				x: FloatType => return curBlock.pure(if(x.is64, V3Op.newFloat64ViewI(oldType), V3Op.newFloat32ViewI(oldType)), [i_scalar]);
+			}
+		}
+		if (IntType.?(newType)) {
+			// XXX: Hack - Something goes wrong with the JVM backend if we use `oneConst()` and `zeroConst()` here
+			var one = if(IntType.!(newType).width > 32, newGraph.valConst(newType, Long.box(1)), newGraph.oneConst());
```

Comment:
You can use the `IntType.box()` method which chooses for you, and then use `SsaGraph.valConst`--it's a little less efficient, but cleaner here. This is probably where we should introduce the `IntViewBool`.

Reply:
Yeah, I'll do `IntViewBool` in a separate PR.